Bug https://bugs.gentoo.org/865361

gcc does not support -fcf-protection for i386 (needs i686+),
so disable it when building iPxeBaseBin.

--- a/src/VBox/Devices/PC/ipxe/Makefile.kmk
+++ b/src/VBox/Devices/PC/ipxe/Makefile.kmk
@@ -176,6 +177,8 @@
 
  iPxeBaseBin_TEMPLATE = iPxe
 
+ iPxeBaseBin_CFLAGS = -fcf-protection=none
+
  iPxeBaseBin_INCS = \
 	src \
 	src/include \
